apple/internal/ios_rules.bzl

```diff
@@ -131,6 +131,7 @@ def _ios_application_impl(ctx):
         ),
         partials.resources_partial(
             bundle_id = bundle_id,
+            bundle_package_type = "APPL",
```

Comment:
There's also a simplification opportunity here. The bundle package type should be associated with the _product type_ rather than the _rule type_. It's a subtle distinction, but since that's the way Xcode models it, it's best to retain that conceptual mapping.

To do that, you can add `bundle_package_type` as a new argument/field to the rule descriptors in rule_support.bzl and associated the appropriate 4-letter code with each of the product types listed there.

Then, these rule functions don't have to change at all. Instead of passing the bundle package type into the partials, have them call `rule_support.rule_descriptor(ctx)` (if they don't already) to retrieve the rule descriptor and pull the package type out of it that way.